lib/src/auth.dart

```diff
@@ -264,3 +282,102 @@ class ReadOnlyAuthenticator extends Authenticator {
     await _requestToken(accountInfo);
   }
 }
+
+/// The [WebAuthenticator] class allows for the creation of an [Authenticator]
+/// that exposes functionality which allows for the user to authenticate through
+/// a browser. The [url] method is used to generate the URL that the user uses
+/// to authenticate on www.reddit.com, and the [authorize] method retrieves the
+/// access token given the returned `code`. This is to be
+/// used with the 'Web' app type credentials. Refer to
+/// https://github.com/reddit/reddit/wiki/OAuth2-App-Types for descriptions of
+/// valid app types.
+class WebAuthenticator extends Authenticator {
+  static WebAuthenticator Create(
```

Comment:
Only type names should begin with uppercase letters in Dart: https://www.dartlang.org/guides/language/effective-dart/style#do-name-types-using-uppercamelcase